libbeat/reader/multiline/message_buffer.go

```diff
@@ -120,6 +123,7 @@ func (b *messageBuffer) addLine(m reader.Message) {
 // finalize writes the existing content into the returned message and resets all reader variables.
 func (b *messageBuffer) finalize() reader.Message {
 	if b.truncated > 0 {
+		b.logger.Warnf("Multiline message is too large, truncated to the limit of %d lines or %d bytes", b.maxLines, b.maxBytes)
```

Comment:
This is potentially a per event log message (if e.g. every message were truncated but that's probably an exception), meaning it could be generated 1000s of times per second.

I recall we had rate limited a log message before, this seems like it could benefit from that. https://github.com/elastic/beats/pull/40448. It depends on how frequent we think this could be.

It would be nice if we had rate limiting and/or sampling and/or aggregation of log messages on the data path by default one day to avoid accidental log spam.

Reply:
Thanks, Craig. I'm revisiting this only now. I think this is a rare case that is most important to give visibility when the multiline option was misconfigured, It should definitely not report for every event. 

I updated the code to only log the message every 1000th message that was truncated, I'm not sure if this is a good approach.

Reply:
Additionally, I have been working on rate limiting support for the logger, appreciate any feedback: https://github.com/elastic/elastic-agent-libs/pull/256